### PR TITLE
Iss58

### PR DESCRIPTION
**Pull request by pbellec**
Replacing NaNs with zeros in the confounds (issue #59) and adding tests regressing out each category of confounds with nilearn (issue #58).

### CODE REVIEW

File: load_confounds/parser.py [hunk 1]
```diff
@@ -332,11 +334,18 @@ def _load_single(self, confounds_raw):
...
             confounds = pd.concat([confounds, confounds_compcor], axis=1)
 
+
+        # Convert from DataFrame to numpy ndarray
         labels = confounds.columns
+        confounds = confounds.values
+
+        # Derivatives have NaN
+        # Replace them by zeros, otherwise nilearn will not regress them out.
+        confounds[np.isnan(confounds)] = 0
```
Comment (arnaudbore) [inline load_confounds/parser.py:344]: confounds = np.nan_to_num(confounds)

Comment (arnaudbore) [reply load_confounds/parser.py:344]: your call it won't change the output

Comment (bpinsard) [reply load_confounds/parser.py:344]: `np.nan_to_num(confounds, copy=False)` 
more efficient

Comment (arnaudbore) [reply load_confounds/parser.py:344]: or
`confounds = np.nan_to_num(confounds, nan=np.mean(confounds))`

Comment (bpinsard) [reply load_confounds/parser.py:344]: not sure the mean of all confounds would make sense, this would had to be column by column

File: load_confounds/parser.py [hunk 2]
```diff
@@ -332,11 +334,18 @@ def _load_single(self, confounds_raw):
...
+
+        # Optionally demean confounds
         if self.demean:
             confounds = scale(confounds, axis=0, with_std=False)
```
Comment (bpinsard) [inline load_confounds/parser.py:348]: If you demean after replacing nans by zeros, these won't be zero anymore. Might be a problem in regression.

Comment (bpinsard) [reply load_confounds/parser.py:348]: `scale(confounds, axis=0, with_std=False,copy=False)` for in-place computation

Comment (bpinsard) [reply load_confounds/parser.py:348]: as scaling is robust to nans (https://github.com/scikit-learn/scikit-learn/blob/fd237278e/sklearn/preprocessing/_data.py#L159) , it would be better to run it before:

File: load_confounds/parser.py (hunk 1 above)
Comment (pbellec) [reply load_confounds/parser.py:344]: thanks for the tip.

File: load_confounds/parser.py (hunk 2 above)
Comment (pbellec) [reply load_confounds/parser.py:348]: @HanadS noted that we are expecting nans only on the first row of the derivatives, so it may make more sense to duplicate the values of the second row in the first row.

Comment (bpinsard) [reply load_confounds/parser.py:348]: It seems to me that it would make more sense to regress with zero values, and not correcting that time point for derivatives rather than guessing that the motion was similar/continuous.

Comment (pbellec) [reply load_confounds/parser.py:350]: ignoring the effect of derivatives in the regression would require to change the way the regression is implemented, in nilearn. Replacing with zero will not exclude the time point (this is a multivariate process), I don't see a work around.

Also even if hypothetically there was a patch submitted to nilearn, it would add a lot of complexity in the code, while the impact of that single time point will likely be very limited.

Comment (bpinsard) [reply load_confounds/parser.py:350]: I don't know which type of regression is used afterwards, I thought this would be a GLM. But sure, this will have limited impact.

Comment (pbellec) [reply load_confounds/parser.py:350]: what's implemented in nilearn is an orthogonal projection with a QR decomposition. I have not thought of the impact setting a particual time point to zero there would have. For a traditional GLM I have not really thought about it, there is the inversion of X'X. Putting a coeff to zero in one of the vectors means that this term will be absent in the coefficients of X'X, but you now have some coefficients with, say, N terms while the other ones have N-1. And then you invert that. Not sure what happens there and if the resulting coefficients are somehow equivalent to ignoring the time point.  

Replacing by the closest value in time seems a relatively safe move to me, as the attribution of the gradient `y[t]-y[t-1]` to `gradient[t]` or `gradient[t-1]` is kind of arbitrary anyway.

File: load_confounds/parser.py (hunk 1 above)
Comment (pbellec) [reply load_confounds/parser.py]: In the current version NaNs are not replaced by zeros anymore. See the thread below.
